src/wallet/wallet.cpp

```diff
@@ -2423,8 +2424,11 @@ bool CWallet::IsSpent(const uint256& hash, unsigned int n) const
     {
         const uint256& wtxid = it->second;
         std::map<uint256, CWalletTx>::const_iterator mit = mapWallet.find(wtxid);
-        if (mit != mapWallet.end() && mit->second.GetDepthInMainChain() >= 0)
-            return true; // Spent
+        if (mit != mapWallet.end()) {
+            auto confirmations = mit->second.GetDepthInMainChain(asOfHeight);
+            if (confirmations >= 0)
+                return true;
+        }
```

Comment:
```suggestion
        if (mit != mapWallet.end() && mit->second.GetDepthInMainChain(asOfHeight) >= 0) {
            return true; // Spent
        }
```

Reply:
IIRC I had added the variable here just to make the code more legible, because it took me some inspection to figure out what was going on in the first place (i.e. why we were checking against the depth in the main chain). However changing this back will make the diff smaller, so :+1: